### ISSUE TITLE
hasSize has been removed from ObjectArrayAssert class

### ISSUE THREAD

**Issue by mipper** [assertj/assertj-core#2131]
#### Summary
The `hasSize(int)` method is no longer available on 2D arrays after commit a36275ecd89ffa12cca931e65b3c9e4fa5322f7f which was released as part of 3.17.0.

I have no idea whether other methods have also become unavailable on this, not whether other array classes are affected.

#### Example
This test will compile under 3.16.1 but not on any version after that up to 3.18.1.

```java
package org.asserj;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class HasSizeTest
{
    @Test
    public void shouldAnswerWithTrue()
    {
        assertThat(new String[][] { { "1" } }).hasSize(1);
    }
}
```

**Comment by scordio** [comment]
@mipper the introduction of [2D array assertions](https://assertj.github.io/doc/#assertj-core-3.17.0-2d-array-assertions) in 3.17.0 is mentioned as a [breaking change](https://assertj.github.io/doc/#assertj-core-3.17.0-breaking-changes).

In the example, `hasSize(1)` should be replaced with `hasDimensions(1, 1)`.

**Comment by mipper** [comment]
I saw the release notes, but they don't list any methods that have been removed.  It's a shame they're not listed along with the methods which should replace them.

**Comment by joel-costigliola** [comment]
@mipper fair enough, would you mind improving the release notes for other users?
